tests/generators/test_binomial_tree.py

```diff
@@ -0,0 +1,53 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+import retworkx
+
+
+class TestBinomailTreeGraph(unittest.TestCase):
+
+    def test_binomial_tree_graph(self):
+        for n in range(10):
+            graph = retworkx.generators.binomial_tree_graph(n)
+            self.assertEqual(len(graph), 2**n)
+            self.assertEqual(len(graph.edges()), 2**n - 1)
```

Comment:
Can we add some assertions here on the edge list. It would be good to assert that the output graph is actually a binomial tree instead of just a graph with `2**n` nodes and `2**n - 1` edges

Reply:
Added assertions to check if the edges match what is expected for a binomial tree graph